### ISSUE TITLE
[Bug] issues_to_lsp_diagnostic returns a wrong link for codeDescription property

### ISSUE THREAD

**Issue by d-biehl** [MarketSquare/robotframework-robocop#937]
### What happened?

Just implementing the code description for `robocop` in `RobotCode`. For new versions of robocop, I think startet with 4.0 the  function [issues_to_lsp_diagnostic](https://github.com/MarketSquare/robotframework-robocop/blob/5187992f3ac675db3f20f75d0f5e5ab209d5979a/robocop/utils/misc.py#L74) returns a wrong link to the description of a rule.

- in robocop 4.0.0 you have included the severity of the rule in the anchor/link in `rules.html`. Because `issues_to_lsp_diagnostic` does not work correctly, I startet to build the link by myself, but if you reconfigure the severity of the rule, then it is not easy possible to get the original severity.

- in robocop 4.1.0 you changed the name of the rules.html to rules_list.html.

I think in SARIF protocol you get also a wrong link, but I did not check this.

### What command/code did you try to run?

 [issues_to_lsp_diagnostic](https://github.com/MarketSquare/robotframework-robocop/blob/5187992f3ac675db3f20f75d0f5e5ab209d5979a/robocop/utils/misc.py#L74)

### What did you expect to happen instead?

the correct link to the rule description.

### Robocop version

4.1, 4.0

**Comment by bhirsz** [comment]
The link is automatically generated from the paragraph name. Like you noticed, since severity is dynamic it can lead to broken urls. Thanks for noticing it. I think we should generate link separetly and not rely on automatic rst links . For example:

```
_ our_custom_link_like_rule_name

Human readable rule name with default severity
-----------------------------------------------------
```

And thanks for pointing out the problem with renaming rules.html -> rules_list.html .  I will check and fix it (though it would be nice to have some tests or more dynamic solution for it.. but I could overthink it since it's first and probably last name we renamed the document name).

**Comment by bhirsz** [comment]
I've tested it and while rules.html was renamed to rules_list.html (I will fix it in next PR), the link to rule itself should still work (for example you can either use https://robocop.readthedocs.io/en/stable/rules_list.html#duplicated-metadata or https://robocop.readthedocs.io/en/stable/rules_list.html#duplicated-metadata-w0806). It's hidden url so it may not be visible from the UI if you hover over paragraph.

**Comment by bhirsz** [comment]
Released in 4.1.1
